cmd/testpeers.go

```diff
@@ -110,6 +292,9 @@ func runTestPeers(ctx context.Context, w io.Writer, cfg testPeersConfig) (err er
 				break
 			}
 			maps.Copy(testResults, result)
+		case <-ctx.Done():
+			selfFinished = true
```

Comment:
That's really neat way of solving that! I have one concern though, reading `sync/errgroup.WithContext` I see:
```
// The derived Context is canceled the first time a function passed to Go
// returns a non-nil error or the first time Wait returns, whichever occurs
// first.
```
For my use case, that means if testing self fails, it will automatically stop my peers tests? If that's the case, it's not something I'd like to happen...

Reply:
If your testing routine got an error, then something is really bad and your test results will be inaccurate anyway, better to yield the error and shutdown.
Or, simply design test routines to never return an error if you wish. You don't have to deal with errors in this case.